src/main/java/org/mondemand/RealTimeClient.java

```diff
@@ -0,0 +1,180 @@
+package org.mondemand;
+
+import java.net.InetAddress;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.log4j.Logger;
+import org.mondemand.transport.LWESTransport;
+
+import com.google.common.util.concurrent.AtomicLongMap;
+
+/**
+ * This is an mondemand client that emit stats according to platformhash, site, aduit
+ *
+ */
+public class RealTimeClient {
+  private static final Logger LOG = Logger.getLogger(RealTimeClient.class);
+
+  protected final String host;
+  protected final int port;
+  protected final Client client;
+  protected final String clientName;
+  private ConcurrentMap<Context, AtomicLongMap<String>>  mondemandContent
+        = new ConcurrentHashMap<Context, AtomicLongMap<String>>();
+  private volatile boolean enabled = true;
+  protected final int INTERVAL = 60000;
+  protected boolean statsReset = true;
+
+  private static RealTimeClient rtClient = null;
+
+  /**
+   * constructor for the singleton
+   * @param clientName
+   * @param host
+   * @param port
+   */
+  protected RealTimeClient(String clientName, String host, int port)
+  {
+    this.clientName = clientName;
+    this.host = host;
+    this.port = port;
+    this.client = createClient();
+  }
+
+  /**
+   * create a singleton for the realtime mondemand client
+   * @param clientName
+   * @param host
+   * @param port
+   * @return the singleton of the RealTimeMondemandClient
+   */
+  public static synchronized RealTimeClient getInstance(String clientName, String host, int port)
+  {
+    if (rtClient == null)
+    {
+      rtClient = new RealTimeClient(clientName, host, port);
+    }
+    return rtClient;
+  }
+
+  private Client createClient()
+  {
+    try
+    {
+      Client newClient = new Client(clientName);
+      newClient.addTransport
+      (new LWESTransport(InetAddress.getByName(host),
+                         port,
+                         null));
+      String hostName = InetAddress.getLocalHost().getHostName();
+      if (hostName.indexOf(".") > 0)
+      {
+        hostName = hostName.substring(0, hostName.indexOf("."));
+      }
+
+      newClient.addContext("host", hostName);
+
+      LOG.info
+      (String.format
+       ("Created mondemand client (%s:%d, client=%s, host=%s)",
+        host,
+        port,
+        clientName,
+        hostName));
+
+      return newClient;
+
+    }
+    catch (Exception e)
+    {
+      e.printStackTrace();
+      LOG.error("System will shutdown due to error: ", e);
+      System.exit(100);
+    }
+
+    return null;
+  }
+
+
+  /**
+   * Increment the count for the map
+   * @param ph : platformhash
+   * @param site : siteId
+   * @param adunit : adunitId
+   * @param key : KeyType: blank, advertiser_revenue, etc
+   * @param value : the value to increase
+   */
+  public void increment(Context context, String keyType, long value)
+  {
+    synchronized (keyType) {
```

Comment:
I don't think you need to use synchronized if you are already using AtomicLongMap